### PR TITLE
util/cteutil: migrate test-infra to testify

### LINKED ISSUE #26173: migrate test-infra to testify for `util/cteutil` pkg

### PR DESCRIPTION
**Pull request by xiaoshengMr**
Signed-off-by: xiaoshengmr <xiaoshengmr@163.com>

### What problem does this PR solve?

Issue Number: close #26173 

### Release note 

-No release note

### CODE REVIEW

File: util/cteutil/storage_test.go [hunk 1]
```diff
@@ -14,78 +14,70 @@
 package cteutil
 
 import (
+	"github.com/stretchr/testify/require"
```
Comment (tisonkun) [inline util/cteutil/storage_test.go:17]: Our imports style follows goimports style, check https://goinbigdata.com/goimports-vs-gofmt/ and I think I should write a section for dev guide.

File: util/cteutil/storage_test.go [hunk 2]
```diff
@@ -14,78 +14,70 @@
...
 	fields := []*types.FieldType{types.NewFieldType(mysql.TypeLong)}
 	chkSize := 1
 	storage := NewStorageRowContainer(fields, chkSize)
-	c.Assert(storage, check.NotNil)
+	require.NotNil(t, storage)
 
 	// Close before open.
 	err := storage.DerefAndClose()
-	c.Assert(err, check.NotNil)
+	require.NotNil(t, err)
```
Comment (tisonkun) [inline util/cteutil/storage_test.go:35]: You may try to use `requuire.NoError` as a more precious checker.

File: util/cteutil/storage_test.go [hunk 3]
```diff
@@ -14,78 +14,70 @@
...
-func TestT(t *testing.T) {
-	check.TestingT(t)
-}
-
-var _ = check.Suite(&StorageRCTestSuite{})
-
-type StorageRCTestSuite struct{}
-
-func (test *StorageRCTestSuite) TestStorageBasic(c *check.C) {
+func TestStorageBasic(t *testing.T) {
```
Comment (tisonkun) [inline util/cteutil/storage_test.go:27]: Think of that whether each test can run in parallel by `t.Parallel()`

File: util/cteutil/storage_test.go [hunk 4]
```diff
@@ -14,78 +14,70 @@
...
-		c.Assert(err, check.IsNil)
+		require.NoError(t, err)
 	}
 	err := storage.DerefAndClose()
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 
 	err = storage.DerefAndClose()
-	c.Assert(err, check.NotNil)
+	require.NotNil(t, err)
```
Comment (tisonkun) [inline util/cteutil/storage_test.go:77]: ditto NoError

File: util/cteutil/storage_test.go [hunk 5]
```diff
@@ -136,101 +128,100 @@ func (test *StorageRCTestSuite) TestSpillToDisk(c *check.C) {
...
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 	outChk, err1 := storage.GetChunk(0)
-	c.Assert(err1, check.IsNil)
+	require.NoError(t, err1)
+
 	in64s := inChk.Column(0).Int64s()
 	out64s := outChk.Column(0).Int64s()
-	c.Assert(reflect.DeepEqual(in64s, out64s), check.IsTrue)
+	require.True(t, reflect.DeepEqual(in64s, out64s))
```
Comment (tisonkun) [inline util/cteutil/storage_test.go:137]: `require.Equal` should be enough if you take a look at their implement.

File: util/cteutil/storage_test.go [hunk 6]
```diff
@@ -136,101 +128,100 @@ func (test *StorageRCTestSuite) TestSpillToDisk(c *check.C) {
...
+	require.Greater(t, memTracker.MaxConsumed(), int64(0))
+	require.Greater(t, diskTracker.BytesConsumed(), int64(0))
+	require.Greater(t, diskTracker.MaxConsumed(), int64(0))
 
 	outChk, err = storage.GetChunk(0)
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 	out64s = outChk.Column(0).Int64s()
-	c.Assert(reflect.DeepEqual(in64s, out64s), check.IsTrue)
+	require.True(t, reflect.DeepEqual(in64s, out64s))
```
Comment (tisonkun) [inline util/cteutil/storage_test.go:156]: ditto and others

**Review by tisonkun — COMMENTED** [review]
Thanks for submitting the PR @xiaoshengMr ! Comments inline. And you can click the CLA button to sign a CLA.

File: util/cteutil/storage_test.go (hunk 2 above)
Comment (xiaoshengMr) [reply util/cteutil/storage_test.go]: This test used check storage can not close before open ,  I think use require.Error may be better

Comment (tisonkun) [reply util/cteutil/storage_test.go]: Oops, you're right. I always make mistake on boolean logic 😆

**Comment by xiaoshengMr** [comment]
@tisonkun  Hi tison , I make a mistake in first commit by error github user can not  compelet license check,  so I have open a new pr(https://github.com/pingcap/tidb/pull/26738) to fix comment problem.  this can be close .

**Comment by tisonkun** [comment]
OK. You can close by yourself. This time let me close it for you.

**Pull request closed without merging**